host/helpers.go

```diff
@@ -9,3 +13,14 @@ func InfoFromHost(h Host) *peer.AddrInfo {
 		Addrs: h.Addrs(),
 	}
 }
+
+// SignedRoutingStateFromHost returns a SignedRoutingState record containing
+// the Host's listen addresses, signed with the Host's private key.
+func SignedRoutingStateFromHost(h Host) (*routing.SignedRoutingState, error) {
```

Comment:
This seems like the place to filter out LAN addresses. I found [WANSharableAddrs in addrutil](https://godoc.org/github.com/libp2p/go-addr-util#WANShareableAddrs), which seems like it should do the trick.

We want to be able to opt-in to including LAN addresses though, and I'm not sure what the best way to do that is, since this is a free function. I guess we could use a global variable, but that seems messy.

Maybe this function should be a method on the `Host` instead, and we can set a flag when constructing the Host instance in go-libp2p? @vyzo, thoughts?

Reply:
BTW, the reason I didn't make this a method on Host in the first place was just to avoid breaking the Host interface. I know we have a few implementations (basichost and the ones in go-libp2p), but maybe downstream users have their own that we don't know about?

Reply:
It can take a boolean for the option, or utilize the functional options pattern.
I doubt anyone else has a host implementation, but it is fine with this being free-standing.

Reply:
hmm, okay, it looks like [we want to deprecate go-addr-util](https://github.com/libp2p/go-addr-util/issues/18) and some digging reveals that it doesn't filter out standard LAN subnets, just loopback and IPv6 link-local addrs. So perhaps the filtering will take a little more work.

Reply:
Check out go-multiaddr-net; we have an `IsPublicAddr` utility function you could use for the filter.

Reply:
thanks, that works 👍